### PR TITLE
GO-2036: attempt to extract objecttypes from state to separate component

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

Extracted object types to separate component according task from linear
### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2036/try-to-extract-objecttypes-from-state-to-separate-component
### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: core/block/editor/state/objecttypes.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	SetObjectTypeKey(objectTypeKey domain.TypeKey)
+	SetNoObjectType(noObjectType bool)
+	GetObjectTypeHistory() *undo.ObjectType
+	SetParentObjectType()
+}
+
+type ObjectTypes struct {
+	objectTypeKeys []domain.TypeKey
+	parent         ObjectType
+	noObjectType   bool
```
Comment (KirillSto) [inline core/block/editor/state/objecttypes.go:37]: Is noObjectType necessary here?

File: core/block/editor/state/state.go [hunk 2]
```diff
@@ -164,11 +165,25 @@ func (s *State) RootId() string {
 }
 
 func (s *State) NewState() *State {
-	return &State{parent: s, blocks: make(map[string]simple.Block), rootId: s.rootId, noObjectType: s.noObjectType, migrationVersion: s.migrationVersion, uniqueKeyInternal: s.uniqueKeyInternal, originalCreatedTimestamp: s.originalCreatedTimestamp}
+	return &State{parent: s,
+		blocks: make(map[string]simple.Block),
+		rootId: s.rootId, migrationVersion: s.migrationVersion,
```
Comment (KirillSto) [inline core/block/editor/state/state.go:170]: Fix indentation for migrationVersion

Comment (AnastasiaShemyakinskaya) [reply core/block/editor/state/state.go]: Fixed

File: core/block/editor/state/objecttypes.go (hunk 1 above)
Comment (AnastasiaShemyakinskaya) [reply core/block/editor/state/objecttypes.go:37]: Yes, it also related to object type component in state. It also used in `ObjectTypeKey()` function

**Pull request closed without merging**